python/analyzers/HNLTreeProducer.py

```diff
@@ -124,20 +188,44 @@ def process(self, event):
         '''
         self.readCollections(event.input)
         self.tree.reset()
-
+        self.event = event
         # event variables 
         self.fillEvent(self.tree, event)
-
-        self.fill(self.tree, 'n_cands', len(event.dimuonsvtx))
+        self.fill     (self.tree, 'n_cands', len(event.dileptonvtx))
+        self.fill     (self.tree, 'rho'    , event.rho)
+        self.fill     (self.tree, 'nLeptons'    , event.nLeptons)
+        self.fill     (self.tree, 'nElectrons'  , event.nElectrons)
+        self.fill     (self.tree, 'nMuons'    , event.nMuons)
 
         # reco HNL
-        self.fillHNL (self.tree, 'hnl'        , event.the_3lep_cand           )
-        self.fill    (self.tree, 'hnl_iso_abs', event.the_3lep_cand.abs_ch_iso)
-        self.fill    (self.tree, 'hnl_iso_rel', event.the_3lep_cand.rel_ch_iso)
-        self.fillMuon(self.tree, 'l1'         , event.the_3lep_cand.l1()      )
-        self.fillMuon(self.tree, 'l2'         , event.the_3lep_cand.l2()      )
-        if self.cfg_ana.promptLepType == 'ele': self.fillEle (self.tree, 'l0', event.the_3lep_cand.l0())
-        if self.cfg_ana.promptLepType == 'mu' : self.fillMuon(self.tree, 'l0', event.the_3lep_cand.l0())
+        self.fillHNL (self.tree, 'hnl'                  , event.the_3lep_cand                 )
+        self.fill    (self.tree, 'hnl_iso03_abs_rhoArea', event.the_3lep_cand.abs_tot_iso03_rhoArea)
+        self.fill    (self.tree, 'hnl_iso04_abs_rhoArea', event.the_3lep_cand.abs_tot_iso04_rhoArea)
+        self.fill    (self.tree, 'hnl_iso05_abs_rhoArea', event.the_3lep_cand.abs_tot_iso05_rhoArea)
+        self.fill    (self.tree, 'hnl_iso03_abs_deltaBeta', event.the_3lep_cand.abs_tot_iso03_deltaBeta)
+        self.fill    (self.tree, 'hnl_iso04_abs_deltaBeta', event.the_3lep_cand.abs_tot_iso04_deltaBeta)
+        self.fill    (self.tree, 'hnl_iso05_abs_deltaBeta', event.the_3lep_cand.abs_tot_iso05_deltaBeta)
+#        self.fill    (self.tree, 'hnl_iso_abs_met'      , event.the_3lep_cand.abs_ch_iso_met  )
+        self.fill    (self.tree, 'hnl_iso03_rel_rhoArea', event.the_3lep_cand.rel_tot_iso03_rhoArea)
+        self.fill    (self.tree, 'hnl_iso04_rel_rhoArea', event.the_3lep_cand.rel_tot_iso04_rhoArea)
+        self.fill    (self.tree, 'hnl_iso05_rel_rhoArea', event.the_3lep_cand.rel_tot_iso05_rhoArea)
+        self.fill    (self.tree, 'hnl_iso03_rel_deltaBeta', event.the_3lep_cand.rel_tot_iso03_deltaBeta)
+        self.fill    (self.tree, 'hnl_iso04_rel_deltaBeta', event.the_3lep_cand.rel_tot_iso04_deltaBeta)
+        self.fill    (self.tree, 'hnl_iso05_rel_deltaBeta', event.the_3lep_cand.rel_tot_iso05_deltaBeta)
```

Comment:
it would be good to move all these single variable `book / fill` to some `bookHNL / fillHNL`

Reply:
yes that's true, maybe we can do it for another version? I left a comment there.

Reply:
ok